pkg/validators/am0005_test_harness.go

```diff
@@ -0,0 +1,81 @@
+// ValidateAddonLabel validates whether the 'label' field under an addon.yaml follows the format 'api.openshift.com/addon-<id>'
+// TODO - remove this validator once we do field level validation
+package validators
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/mt-sre/addon-metadata-operator/pkg/utils"
+	imageparser "github.com/novln/docker-parser"
+)
+
+const quayRegistryApi = "https://quay.io/v2"
+
+var AM0005 = utils.Validator{
+	Code:        "AM0005",
+	Name:        "test_harness",
+	Description: "Ensure that an addon has a valid testharness image",
+	Runner:      ValidateTestHarness,
+}
+
+func init() {
+	Registry.Add(AM0005)
+}
+
+func ValidateTestHarness(metabundle utils.MetaBundle) (bool, string, error) {
+	res, err := imageparser.Parse(metabundle.AddonMeta.TestHarness)
+	if err != nil {
+		return false, "Failed to parse testharness url", err
+	}
+	if res.Registry() != "quay.io" {
+		return false, "Testharness image is not in the quay.io registry", nil
+	}
+	imagePresent, err := checkImageExists(res, 5)
```

Comment:
nit: add a var named `retryCount` so we understand the code easily -> `checkImageExists(res, retryCount)`